src/main/java/uk/gov/companieshouse/api/accounts/service/impl/FilingServiceImpl.java

```diff
@@ -0,0 +1,371 @@
+    private static final String IXBRL_LOCATION = "s3://%s/%s/%s";
+    private static final String LINK_RELATIONSHIP = "accounts";
+    private static final String PERIOD_END_ON = "period_end_on";
+
+    private final EnvironmentReader environmentReader;
+    private final ObjectMapper objectMapper;
+    private final IxbrlGenerator ixbrlGenerator;
+    private final AccountsBuilder accountsBuilder;
+
+    @Autowired
+    public FilingServiceImpl(
+        EnvironmentReader environmentReader,
+        ObjectMapper objectMapper,
+        IxbrlGenerator ixbrlGenerator,
+        AccountsBuilder accountsBuilder) {
+
+        this.objectMapper = objectMapper;
+        this.environmentReader = environmentReader;
+        this.ixbrlGenerator = ixbrlGenerator;
+        this.accountsBuilder = accountsBuilder;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Filing generateAccountFiling(Transaction transaction, CompanyAccountEntity accountEntity)
+        throws IOException {
+        Filing filing = null;
+
+        if (isSmallFullAccount(accountEntity)) {
+            filing = generateAccountFiling(transaction, AccountsType.SMALL_FULL_ACCOUNTS);
+        }
+
+        return filing;
+    }
+
+    /**
+     * Generate the filing for the account type passed in.
+     *
+     * @param transaction - transaction id
+     * @param accountsType - Account type information: account type, ixbrl's template name, account
+     * @return {@link Filing} - null or filing with the filing information (e.g. ixbrl location,
+     * accounts name, etc)
+     * @throws IOException -
+     */
+    private Filing generateAccountFiling(Transaction transaction, AccountsType accountsType)
+        throws IOException {
+        Filing filing = null;
+        Object accountObj = getAccountInformation(accountsType.getAccountType());
+
+        if (accountObj != null) {
+            String ixbrlLocation = callIxbrlGenerator(accountsType,
+                generateJson(accountsType.getAccountType(), accountObj));
+
+            if (ixbrlLocation != null && isValidIXBL()) {
+                filing = createAccountFiling(transaction, accountsType, ixbrlLocation);
+            }
+        }
+
+        return filing;
+    }
+
+    /**
+     * Generate the json for the passed-in account. This is used in the request body when calling
+     * the service.
+     *
+     * @param accountType - Account type information (e.g. small-full, abridged, etc)
+     * @param accountObj - The account's type object that's being processed (e.g. Account)
+     * @return null or ixbrl location
+     * @throws JsonProcessingException
+     */
+    private String generateJson(String accountType, Object accountObj)
+        throws JsonProcessingException {
+        if (isSmallFullType(accountType)) {
+            return generateCompanyAccountJSON((Account) accountObj);
+        }
+        return null;
```

Comment:
Logging needs to be added to be able to diagnose issues. This if statement could be missed when future account types are added and logging would help developers quickly find the problem.

Reply:
commit: 4a47945